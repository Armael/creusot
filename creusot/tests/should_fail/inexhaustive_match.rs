fn inexhaustive_match(x: Option<()>) {
    match x {
        None => (),
    }
}
