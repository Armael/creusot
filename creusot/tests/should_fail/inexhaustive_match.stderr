error[E0004]: non-exhaustive patterns: `Some(_)` not covered
 --> inexhaustive_match.rs:2:11
  |
2 |     match x {
  |           ^ pattern `Some(_)` not covered
  |
note: `std::option::Option<()>` defined here
  = note: the matched value is of type `std::option::Option<()>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
  |
3 ~         None => (),
4 ~         Some(_) => todo!(),
  |

warning: load the `creusot_contract` crate to enable resolution of mutable borrows.

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0004`.
